intermediate-js.md

```diff
@@ -3,37 +3,39 @@
 ### Without using Google answer the following:
 
 1. What is the difference between .map() and .filter()?
-
+Map and filter are both higher order functions.  The map function is used when you want to manipulate each item in an array while the filter function is used to extract specific data from an array.
 2. Why would you use object destructuring?
-
+Object destructuring is very handy because it allows you to essentially shorthand while writing new variables.
 3. What is the difference between var, let, and const?
-
+Var/let/const are all used to define variables.  "You should always use const when defining a variable, unless you have a good reason not to." -Matt.  Const will keep a variable the same, if you use let, you can redefine the variable again in the future.
 4. Why is testing important?
-
+Testing is important because you can make sure that all of your functions are working before you write a ton of code that won't work!
 5. What is a higher order function?
-
+A higher order function is one that either takes in another function or returns one (map,filter).
 6. What is the difference between a class and an object?
-
+Objects in JavaScript are essentially data storage while classes are declared in a constructor and called on in functions.
```

Comment:
I like the distinction you're making here. You'll want to also keep in mind that classes are often blueprints for objects and that objects are often instances of classes.